rust/osm2lanes/src/transform/tags_to_lanes/bus.rs

```diff
@@ -55,84 +70,101 @@ pub(super) fn bus(
     Ok(())
 }
 
-fn busway(
-    tags: &Tags,
-    locale: &Locale,
-    road: &mut RoadBuilder,
-    _warnings: &mut RoadWarnings,
-) -> Result<(), RoadError> {
-    const BUSWAY: TagKey = TagKey::from("busway");
-    if tags.is(BUSWAY, "lane") {
-        road.forward_outside_mut()
-            .ok_or_else(|| RoadError::unsupported_str("no forward lanes for busway"))?
-            .set_bus(locale)?;
-        if !tags.is("oneway", "yes") && !tags.is("oneway:bus", "yes") {
-            road.backward_outside_mut()
-                .ok_or_else(|| RoadError::unsupported_str("no backward lanes for busway"))?
-                .set_bus(locale)?;
+impl BuswayScheme {
+    pub(super) fn new(
+        tags: &Tags,
+        locale: &Locale,
+        oneway: &Oneway,
+        warnings: &mut RoadWarnings,
+    ) -> Self {
+        let mut busway = Self {
+            forward_direction: Infer::Default(None),
+            backward_direction: Infer::Default(None),
+        };
+
```

Comment:
The tiniest of style nitpicks: just one blank newline? cargo fmt may fix for you

Reply:
I have my IDE setup to run cargo format, but it produces different results to `cargo fmt` on the command line (maybe due to the config specifying nightly?). Now I know to run `cargo fmt` before commiting.

Reply:
there is a rustfmt config file, and a toolchain file, that should correctly configure your rust setup. not sure why your IDE might ignore it.